webctrl/script/api_webctrl.py

```diff
@@ -10,15 +10,41 @@
 from sqlalchemy.orm import sessionmaker
 
 # import json #used if we want to output json file
+import argparse
 import configparser
 import logging
+import logging.handlers
 import orm_webctrl
 import os
 import pendulum
 import requests
 
 
-logging.basicConfig(filename='error.log', format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+SCRIPT_NAME = os.path.basename(__file__)
+# parser for script arguments
+parser = argparse.ArgumentParser(description='Get reading data from egauge api and insert into database.')
+#--verbose argument is True if set
+parser.add_argument('-v', '--verbose', action='store_true',
+                    help='print INFO level log messages to console and error.log')
+args = parser.parse_args()
+
+# set log level to INFO only if verbose is set
+if args.verbose:
+    log_level = 'INFO'
+else:
+    log_level = 'ERROR'
+
+# configure logger which will print log messages to console (only prints ERROR level messages by default; prints INFO level messages if --verbose flag is set)
+logging.basicConfig(level=log_level, format=__file__+': %(message)s')
+
+# Create a handler that writes log messages to error.log file
+# rotates error.log every time it reaches 100 MB to limit space usage; keeps up to 5 old error.log files
+rotating_file_handler = logging.handlers.RotatingFileHandler('error.log', maxBytes=100000000, backupCount=5)
+# set the message and date format of file handler
+formatter = logging.Formatter(fmt='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+rotating_file_handler.setFormatter(formatter)
+# add the handler to the root logger
+logging.getLogger('').addHandler(rotating_file_handler)
```

Comment:
This needs to become a function. We should also discuss a bit later tonight on the overall code modules, now that we are more or less at this point.